pkg/operator/controller/controller.go

```diff
@@ -168,11 +170,21 @@ func (r *reconciler) enforceEffectiveIngressDomain(ci *operatorv1.IngressControl
 	updated := ci.DeepCopy()
 	switch {
 	case len(ci.Spec.Domain) > 0:
+		unique, err := r.isDomainUnique(updated)
```

Comment:
`IsDomainUnique` should take a parameter for the domain instead of for the ingresscontroller.